test/unit/org/apache/cassandra/tools/ToolRunner.java

```diff
@@ -52,15 +50,17 @@
     protected static final Logger logger = LoggerFactory.getLogger(ToolRunner.class);
 
     private static final ImmutableList<String> DEFAULT_CLEANERS = ImmutableList.of("(?im)^picked up.*\\R");
-    
+    private static final String[] EMPTY_STRING_ARRAY = new String[0];
+
     private final List<String> allArgs = new ArrayList<>();
     private Process process;
+    @SuppressWarnings("resource")
     private final ByteArrayOutputStream errBuffer = new ByteArrayOutputStream();
+    @SuppressWarnings("resource")
     private final ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
+    private final CompletableFuture<Void> onComplete = new CompletableFuture<>();
```

Comment:
This looks unnecessary now. It's just a wrapper for `stdin.close()` and I don't see it involved in any async work correct? If true just remove it imo and use `stdin.close()` instead